js/basket-signup.js

```diff
@@ -0,0 +1,78 @@
+// Posting to Basket
+// Doc: https://basket.readthedocs.io/newsletter_api.html#news-subscribe
+
+var url = `https://www.mozilla.org/en-US/newsletter/`;
+
+var basketSignup = function(transaction, onSuccessCallback, onFailCallback) {
+  console.log(`basketSignup`, transaction);
+
+  var payload = {
+    format: `H`, // HTML emails
+    newsletter: `mozilla-foundation`,
+    triggerWelcome: `N`,
+    email: transaction.email,
+    privacy: transaction.privacy || false
+  };
+
+  var errorArray = [];
+  var newsletterErrors = [];
+
+  while (newsletterErrors.firstChild) {
+    newsletterErrors.removeChild(newsletterErrors.firstChild);
+  }
+
+  var params =
+    `email=` +
+    encodeURIComponent(payload.email) +
+    `&newsletters=` +
+    payload.newsletter +
+    `&privacy=` +
+    payload.privacy +
+    `&fmt=` +
+    payload.format +
+    `&source_url=` +
+    /* Strip query params in source url for newsletter signups: https://github.com/mozilla/foundation.mozilla.org/issues/2994#issuecomment-516997473 */
+    encodeURIComponent(window.location.href.split(`?`)[0]);
+
+  var xhr = new XMLHttpRequest();
```

Comment:
This code is fairly old, and predates the Fetch API, which is a lot cleaner - would you be up for rewriting it to use `fetch(...).then(...).catch(...)` instead?